src/main/java/org/jabref/logic/importer/fileformat/MarcXmlParser.java

```diff
@@ -398,6 +404,45 @@ private void putNotes(BibEntry bibEntry, Element datafield) {
         }
     }
 
+    private void putIssue(BibEntry bibEntry, Element datafield) {
+        bibEntry.setType(StandardEntryType.Article);
+
+        List<String> issues = getSubfields("g", datafield);
+
+        for (String issue : issues) {
+            String[] parts = issue.split(":");
+            if (parts.length == 2) {
+                String key = parts[0].trim();
+                String value = parts[1].trim();
+
+                if (StringUtil.isNotBlank(value)) {
+                    switch (key) {
+                        case "number" -> bibEntry.setField(StandardField.NUMBER, value);
+                        case "year" -> bibEntry.setField(StandardField.YEAR, value);
+                        case "pages" -> bibEntry.setField(StandardField.PAGES, value);
+                        case "volume" -> bibEntry.setField(StandardField.VOLUME, value);
+                        case "day" -> bibEntry.setField(StandardField.DAY, value);
+                        case "month" -> bibEntry.setField(StandardField.MONTH, value);
+                    }
+                }
+            }
+        }
+    }
+
+    private void putThesisDescription(BibEntry bibEntry, Element datafield) {
+        String thesisDegree = getSubfield("b", datafield);
+        String school = getSubfield("c", datafield);
+        bibEntry.setType(StandardEntryType.MastersThesis);
+
+        if (StringUtil.isNotBlank(school)) {
+            bibEntry.setField(StandardField.SCHOOL, school);
+        }
+        if ("Dissertation".equals(thesisDegree) && StringUtil.isNotBlank(school)) {
```

Comment:
Isn't the check for `Dissertation` enough? School ist set in line 483, so the setting in 442 can be removed

Reply:
Fixed (line 438*)